admin/javascript/LeftAndMain.EditForm.js

```diff
@@ -77,23 +85,44 @@
 					}
 				}
 
+				// Reset error display
+				this.setValidationErrorShown(false);
+
 				// TODO
 				// // Rewrite # links
 				// html = html.replace(/(<a[^>]+href *= *")#/g, '$1' + window.location.href.replace(/#.*$/,'') + '#');
 				// 
 				// // Rewrite iframe links (for IE)
 				// html = html.replace(/(<iframe[^>]*src=")([^"]+)("[^>]*>)/g, '$1' + $('base').attr('href') + '$2$3');
-				
-				// Show validation errors if necessary
-				if(this.hasClass('validationerror')) {
-					// Ensure the first validation error is visible
-					var tabError = this.find('.message.validation, .message.required').first().closest('.tab');
-					$('.cms-container').clearCurrentTabState(); // clear state to avoid override later on
-					tabError.closest('.ss-tabset').tabs('option', 'active', tabError.index('.tab'));
-				}
 
 				this._super();
 			},
+			'from .cms-tabset': {
```

Comment:
I think you already figured it out in #5412, but in case anyone else wonders (there aren’t any docs for this functionality!) - every time the `afterredrawtabs` event is triggered on `.cms-tabset`, we run the `onafterredrawtabs` event handler in the context of `.cms-edit-form`.

It’s essentially equivalent to putting all of that logic in a function named `handleValidationError` and calling `$('.cms-edit-form').handleValidationError()` at the end of the tab redraw logic. Obviously if you wanted 15 different functions from 15 different selectors to trigger after each tab redraw, that would get pretty ugly and hard to maintain, so triggering an event that other selectors can bind to is much cleaner :)